### ISSUE TITLE
About training objective: is the reference and the target image the same?

### ISSUE THREAD

**Issue by feiyangsuo** [tencent-ailab/IP-Adapter#94]
Thanks for your great work!

I have some questions about the training data details.
In practice, with the provided code, the model can gerenate a whole body with only a face given. And this kind of "text-face-body" pair seems never shown in the training samples. It's like magic to me.

In the paper it says you used text-image pairs as traing data. And in the code the ```MyDataset``` class returns "image" and "clip_image", which are from the same image file. 
After reading the code I infer that during training, the content of the reference image and of the target image to-be-diffused are the same. I can understand that this mechanism helps the model learn to mimic the reference, yet I wonder how do you prevent the model from honestly re-producing the reference image and ignoring the text prompt?

**Comment by xiaohu2015** [comment]
@feiyangsuo hi, for the IP-Adapter in the paper, we use the same image as reference and the target image. As the image condition contains more information to text, hence the model is likely to ignore the text prompt. But as IP-Adapter is 
lightweight adapter and we don't train the original unet, the original model don't loss text prompt ability, we can lower the scale of the image features. And we found that setting with scale=0.5 can make a good balance between text prompt and image prompt for most cases. Besides, we also trained a light IP-Adapter, which we random crop part of image as condition, This alleviates the problem to a certain extent. For the IP-Adapter-Face, we use cropped face image as reference image, hence It basically does not have the above problems
